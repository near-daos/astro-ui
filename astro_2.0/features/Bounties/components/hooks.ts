import { useCallback, useRef, useState } from 'react';
import { useRouter } from 'next/router';
import { useAsyncFn, useLocalStorage, useLocation } from 'react-use';
import axios, { CancelTokenSource } from 'axios';

import { SputnikHttpService } from 'services/sputnik';
import { useModal } from 'components/modal';
import { ConfirmActionModal } from 'astro_2.0/components/ConfirmActionModal';

import { Bounty, BountyContext, BountyProposal } from 'types/bounties';
import { VoteAction } from 'types/proposal';

import { NOTIFICATION_TYPES, showNotification } from 'features/notifications';

import { useWalletContext } from 'context/WalletContext';
import { PaginationResponse } from 'types/api';
import { VOTE_ACTION_SOURCE_PAGE } from 'constants/votingConstants';

export function useBountyControls(
  daoId: string,
  bounty?: Bounty
): {
  handleClaim: () => void;
  handleUnclaim: () => void;
  loading: boolean;
} {
  const router = useRouter();
  const { nearService } = useWalletContext();
  const [, setVoteActionSource] = useLocalStorage(VOTE_ACTION_SOURCE_PAGE);
  const { pathname } = useLocation();
  const [loading, setLoading] = useState(false);

  const [showModal] = useModal(ConfirmActionModal);

  const onSuccessHandler = useCallback(async () => {
    router.reload();
  }, [router]);

  const handleClaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Claim',
    });

    if (res?.length && bounty) {
      setLoading(true);

      const dao = await SputnikHttpService.getDaoById(daoId);

      if (!dao) {
        return;
      }

      try {
        setVoteActionSource(pathname);

        await nearService?.claimBounty(daoId, {
          bountyId: bounty?.bountyId,
          deadline: bounty?.maxDeadline,
          bountyBond: dao.policy.bountyBond,
          gas: res[0],
          tokenId: bounty?.token,
        });

        await onSuccessHandler();
      } catch (err) {
        setLoading(false);
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          lifetime: 5000,
          description: err.message,
        });
      }
    }
  }, [
    showModal,
    bounty,
    daoId,
    setVoteActionSource,
    pathname,
    nearService,
    onSuccessHandler,
  ]);

  const handleUnclaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Unclaim',
    });

    if (res?.length && bounty) {
      try {
        setLoading(true);
        setVoteActionSource(pathname);
        await nearService?.unclaimBounty(daoId, bounty?.bountyId);

        await onSuccessHandler();
      } catch (err) {
        setLoading(false);
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          lifetime: 5000,
          description: err.message,
        });
      }
    }
  }, [
    showModal,
    bounty,
    setVoteActionSource,
    pathname,
    nearService,
    daoId,
    onSuccessHandler,
  ]);

  return {
    handleClaim,
    handleUnclaim,
    loading,
  };
}

export function useBountyVoting(
  daoId: string,
  proposal: BountyProposal
): {
  handleVote: (vote: VoteAction) => void;
  loading: boolean;
} {
  const router = useRouter();
  const [showModal] = useModal(ConfirmActionModal);
  const { nearService } = useWalletContext();
  const [, setVoteActionSource] = useLocalStorage(VOTE_ACTION_SOURCE_PAGE);
  const { pathname } = useLocation();

  const [{ loading }, handleVote] = useAsyncFn(
    async (vote: VoteAction) => {
      const res = await showModal({
        title: 'Confirm Your Vote',
      });

      if (res?.length) {
        setVoteActionSource(pathname);
        await nearService?.vote(daoId, proposal.proposalId, vote, res[0]);
        await router.reload();
      }
    },
    [daoId, proposal, router, nearService, pathname, setVoteActionSource]
  );

  return {
    handleVote,
    loading,
  };
}

export function useBountySearch(): {
  handleSearch: (
    val: string
  ) => Promise<PaginationResponse<BountyContext[]> | null>;
  loading: boolean;
} {
  const router = useRouter();
  const daoId = router.query.dao as string;
  const { accountId } = useWalletContext();
  const cancelTokenRef = useRef<CancelTokenSource | null>(null);

  const [{ loading }, handleSearch] = useAsyncFn(async query => {
    if (cancelTokenRef.current) {
      cancelTokenRef.current?.cancel('Cancelled by new req');
    }

    const { CancelToken } = axios;
    const source = CancelToken.source();

    cancelTokenRef.current = source;

    return SputnikHttpService.findBountyContext({
      daoId: daoId || '',
      accountId,
      query,
      cancelToken: source.token,
    });
  }, []);

  return {
    handleSearch,
    loading,
  };
}

export function useHideBounty(): {
  handleSubmit: () => void;
  handleSelect: (id: string) => void;
  selected: string[];
  loading: boolean;
} {
  const router = useRouter();
  const showHidden = router.query?.bountyFilter === 'hidden';
  const daoId = router.query.dao as string;
  const { accountId, nearService, pkAndSignature } = useWalletContext();
  const [selected, setSelected] = useState<string[]>([]);

  const [{ loading }, handleSubmit] = useAsyncFn(async () => {
    if (!pkAndSignature) {
      return;
    }

    try {
      const { publicKey, signature } = pkAndSignature;

      if (publicKey && signature) {
        await SputnikHttpService.toggleBountyContexts({
          accountId,
          publicKey,
          signature,
          daoId,
          ids: selected,
          isArchived: !showHidden,
        });

        await router.replace(router.asPath, undefined, {
          shallow: false,
        });

        setSelected([]);
      }
    } catch (err) {
      showNotification({
        type: NOTIFICATION_TYPES.ERROR,
        lifetime: 20000,
        description: err.message,
      });
    }
  }, [
    selected,
    daoId,
    showHidden,
    accountId,
    router,
    nearService,
    pkAndSignature,
  ]);

  const handleSelect = useCallback(
    (id: string) => {
      if (selected.includes(id)) {
        setSelected(selected.filter(item => item !== id));
      } else {
        setSelected([...selected, id]);
      }
    },
    [selected]
  );

  return {
    handleSubmit,
    handleSelect,
    selected,
    loading,
  };
}
