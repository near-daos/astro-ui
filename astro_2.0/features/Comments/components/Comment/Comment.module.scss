@import 'styles/typography';

.root {
  background-color: var(--color-white);
  border-radius: 8px;
  box-shadow: 3px 2px 24px rgba(0, 0, 0, 0.1);
  display: flex;
  flex-direction: column;
  margin-bottom: 24px;
  padding: 24px 24px 12px;
  position: relative;
}

.footer {
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin-top: 22px;
}

.replies {
  align-items: center;
  color: var(--color-neutral-70);
  display: flex;

  .repliesToggle {
    align-items: center;
    display: flex;
    justify-content: flex-start;
    padding: 0;
  }

  .toggleIcon {
    margin-left: 8px;
    width: 24px;
  }
}

.replies:hover {
  .toggleIcon {
    color: var(--color-primary-50);
  }
}

.right {
  align-items: center;
  display: flex;
  margin-left: auto;
}

.dislike {
  transform: rotateX(180deg);
}

.delete,
.edit {
  color: var(--color-neutral-60);
  margin-right: 24px;

  &:hover {
    color: var(--color-primary-50);
  }
}

.replyButton {
  margin-left: 24px;
}

.repliesWrapper {
  margin: 12px 0 0 48px;
}

.reply {
  border-top: 1px solid var(--color-neutral-40);
  padding-top: 18px;
}

.replyInputWrapper {
  margin-top: 24px;
}
