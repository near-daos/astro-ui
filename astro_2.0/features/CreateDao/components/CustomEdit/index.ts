import CustomEdit from './CustomEdit';

export default CustomEdit;
