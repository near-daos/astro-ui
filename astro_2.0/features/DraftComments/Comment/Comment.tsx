import React, { FC } from 'react';

import styles from './Comment.module.scss';

export const Comment: FC = () => {
  return <div className={styles.comment}>Comment</div>;
};
