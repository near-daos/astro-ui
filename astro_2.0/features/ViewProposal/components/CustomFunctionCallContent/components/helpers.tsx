import { useMemo } from 'react';
import { configService } from 'services/ConfigService';

export interface StreamInfo {
  amount: string;
  description: string;
  isAutoStartEnabled: boolean;
  ownerId: string;
  receiverId: string;
  tokensPerSec: string;
}

export function useRoketoStreamCheck(json: string): {
  create: boolean;
  stream: StreamInfo;
} {
  const parsedJson = useMemo(() => {
    try {
      return JSON.parse(json);
    } catch (e) {
      return null;
    }
  }, [json]);
  const message = useMemo(() => {
    try {
      return JSON.parse(parsedJson?.msg ?? '{}');
    } catch (e) {
      console.error(e);

      return null;
    }
  }, [parsedJson]);
  const config = configService.get();
  const streamingContract = config.appConfig.ROKETO_CONTRACT_NAME;

  const isStreaming = parsedJson?.receiver_id === streamingContract;
  const request = message?.Create?.request;

  return {
    stream: {
      amount: parsedJson?.amount ?? '0',
      description: request?.description ?? '',
      isAutoStartEnabled: request?.is_auto_start_enabled ?? true,
      ownerId: request?.owner_id ?? '',
      receiverId: request?.receiver_id ?? '',
      tokensPerSec: request?.tokens_per_sec ?? '0',
    },
    create: isStreaming && typeof request !== 'undefined',
  };
}
