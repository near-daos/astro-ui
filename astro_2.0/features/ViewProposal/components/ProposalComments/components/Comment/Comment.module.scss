@import 'styles/typography';
@import 'styles/mixins';

.root {
  @extend %body2;
  background-color: var(--color-neutral-20);
  border-radius: 0 4px 4px;
  color: var(--color-black);
  height: fit-content;
  margin-bottom: 42px;
  //margin-top: auto;
  max-width: 60%;
  padding: 12px;
  position: relative;
  width: fit-content;

  &:first-of-type {
    margin-top: auto;
  }

  p {
    margin: 2px 0;
  }

  .time {
    @extend %caption1;
    display: flex;
    justify-content: flex-end;
  }

  &.myComment {
    background-color: var(--color-neutral-30);
    border-radius: 4px 0 4px 4px;
    margin-left: auto;
  }
}

.tickMark {
  position: absolute;
  top: 0;
}

.tickRight {
  right: -8px;
}

.tickLeft {
  left: -10px;
}

.commentAuthor {
  @extend %title3;
  color: var(--color-primary-50);
}

.commentControls {
  align-items: center;
  display: flex;
  left: 0;
  min-width: 100%;
  position: absolute;
  top: 100%;
  visibility: hidden;
}

.root.myComment {
  .commentControls {
    left: unset;
    right: 0;
  }
}

.commentControlButtonWrapper {
  @include centralize;
  background-color: var(--color-primary-20);
  border-radius: 50%;
  height: 24px;
  transform: translateY(4px);
  width: 24px;

  &:not(:first-of-type) {
    margin-left: 8px;
  }
}

.commentControlButton + div {
  white-space: nowrap;
}

.commentControlIcon {
  height: 16px;
  transform: translateY(-1px);
  width: 16px;
}

.root:hover {
  .commentControls {
    visibility: visible;
  }
}

.commentControls.reported {
  opacity: 0.7;
  pointer-events: none;
  visibility: visible;
}

.reportedInfo {
  @extend %body2;
  margin-left: 4px;
  transform: translateY(4px);
  white-space: nowrap;
}
