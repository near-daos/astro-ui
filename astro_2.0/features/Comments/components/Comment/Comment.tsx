import React, { FC, useState, useCallback } from 'react';
import { useTranslation } from 'next-i18next';

import { DraftComment } from 'services/DraftsService/types';

import { Icon } from 'components/Icon';
import { Button } from 'components/button/Button';
import { CommentContent } from 'astro_2.0/features/Comments/components/CommentContent';
import { EditableContent } from 'astro_2.0/components/EditableContent';
import { ReplyButton } from 'astro_2.0/components/ReplyButton';
import { CommentActions } from 'astro_2.0/features/Comments/components/CommentActions';
import { LikeButton } from 'astro_2.0/features/Comments/components/Comment/LikeButton';

import { useWalletContext } from 'context/WalletContext';

import styles from './Comment.module.scss';

export interface CommentProps {
  data: DraftComment;
  onLike: (id: string, isLiked: boolean) => Promise<void>;
  onDislike: (id: string, isDisliked: boolean) => Promise<void>;
  onReply: (value: string, replyTo: string) => Promise<void>;
  onEdit: (value: string, id: string) => Promise<void>;
  onDelete: (id: string) => Promise<void>;
  isEditable: boolean;
  disabled?: boolean;
}

export const Comment: FC<CommentProps> = ({
  data,
  onLike,
  onDislike,
  onReply,
  onEdit,
  onDelete,
  isEditable,
  disabled,
}) => {
  const {
    author,
    message,
    createdAt,
    updatedAt,
    replies,
    likeAccounts,
    dislikeAccounts = [],
    id,
  } = data;
  const { t } = useTranslation();
  const [edit, setEdit] = useState(false);
  const [contentHtml, setContentHtml] = useState(message);

  const [html, setHTML] = useState('');
  const [showReplies, setShowReplies] = useState(false);
  const [showReplyInput, setShowReplyInput] = useState(false);

  const { accountId } = useWalletContext();
  const isLikedByMe = likeAccounts.includes(accountId);
  const isDislikedByMe = dislikeAccounts.includes(accountId);

  const handlerReply = useCallback(
    msg => {
      onReply(msg, id);

      setHTML('');
      setShowReplies(true);
    },
    [id, onReply]
  );

  const handlerEdit = useCallback(
    msg => {
      onEdit(msg, id);

      setContentHtml('');
      setEdit(false);
    },
    [id, onEdit]
  );

  return (
    <div className={styles.root}>
      <CommentActions
        id={id}
        isEditable={isEditable}
        onEdit={() => setEdit(!edit)}
        onDelete={onDelete}
      />
      {edit ? (
        <EditableContent
          id={`key-edit-${id}`}
          html={contentHtml}
          setHTML={setContentHtml}
          handleSend={handlerEdit}
          handleCancel={() => setEdit(!edit)}
          placeholder={t('drafts.comments.placeholder')}
        />
      ) : (
        <CommentContent
          author={author}
          updatedAt={updatedAt}
          createdAt={createdAt}
          text={message}
        />
      )}
      <div className={styles.footer}>
        {!!replies?.length && (
          <div className={styles.replies}>
            <Button
              className={styles.repliesToggle}
              capitalize
              size="small"
              variant="tertiary"
              onClick={() => setShowReplies(!showReplies)}
            >
              {replies?.length}{' '}
              {replies?.length === 1
                ? t('drafts.comments.reply')
                : t('drafts.comments.replies')}{' '}
              <Icon
                name={showReplies ? 'buttonArrowUp' : 'buttonArrowDown'}
                className={styles.toggleIcon}
              />
            </Button>
          </div>
        )}
        <div className={styles.right}>
          <LikeButton
            disabled={!accountId || Boolean(disabled)}
            onClick={() => onLike(id, isLikedByMe)}
            isActive={isLikedByMe}
            amount={likeAccounts.length}
          />
          <LikeButton
            iconClassName={styles.dislike}
            disabled={!accountId || Boolean(disabled)}
            onClick={() => onDislike(id, isDislikedByMe)}
            isActive={isDislikedByMe}
            amount={dislikeAccounts.length}
          />
          {accountId && (
            <ReplyButton
              title={t('drafts.comments.reply')}
              disabled={disabled}
              className={styles.replyButton}
              onClick={() => setShowReplyInput(!showReplyInput)}
            />
          )}
        </div>
      </div>
      {showReplies && !!replies?.length && (
        <div className={styles.repliesWrapper}>
          {replies?.map(item => (
            <CommentContent
              key={item.id}
              author={item.author}
              createdAt={item.createdAt}
              updatedAt={item.updatedAt}
              text={item.message}
              className={styles.reply}
            />
          ))}
        </div>
      )}
      {showReplyInput && !disabled && (
        <div className={styles.replyInputWrapper}>
          <EditableContent
            id={`key-${id}`}
            html={html}
            setHTML={setHTML}
            handleSend={handlerReply}
            handleCancel={() => setShowReplyInput(!showReplyInput)}
            placeholder={t('drafts.comments.replyPlaceholder')}
          />
        </div>
      )}
    </div>
  );
};
