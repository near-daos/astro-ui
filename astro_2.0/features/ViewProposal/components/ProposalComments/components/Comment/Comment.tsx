import React, { FC, useCallback, useEffect, useRef, useState } from 'react';
import cn from 'classnames';
import { motion } from 'framer-motion';
import { useTranslation } from 'next-i18next';
import Linkify from 'react-linkify';
import { useMountedState } from 'react-use';

import { IconName } from 'components/Icon';
import { ActionButton } from 'astro_2.0/components/ActionButton';
import { ConfirmCommentActionModal } from 'astro_2.0/features/ViewProposal/components/ProposalComments/components/ConfirmCommentActionModal';
import { getImagesFromLinks } from 'astro_2.0/features/ViewProposal/components/ProposalComments/components/Comment/helpers';

import { ReportCommentsInput } from 'types/proposal';

import { useModal } from 'components/modal';

import { formatISODate } from 'utils/format';

import styles from './Comment.module.scss';

interface CommentProps {
  isMyComment: boolean;
  accountId: string;
  message: string;
  isCouncilUser: boolean;
  isReported: boolean;
  reportsCount: number;
  createdAt: string;
  onReport: (params: ReportCommentsInput) => void;
  onDelete: (commentId: number, reason: string) => void;
  commentId: number;
}

const TICK_RIGHT = (
  <svg
    className={cn(styles.tickMark, styles.tickRight)}
    width="12"
    height="12"
    viewBox="0 0 10 12"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1.00034 0C1.00034 0 6.73828 0 8.20033 0C9.66239 0 10.0003 1.5 8.65031 3C7.30031 4.5 1.4997 9.5 1.00034 11C0.500977 12.5 1.00034 0 1.00034 0Z"
      fill="#F0F0F0"
    />
  </svg>
);

const TICK_LEFT = (
  <svg
    className={cn(styles.tickMark, styles.tickLeft)}
    width="12"
    height="12"
    viewBox="0 0 10 12"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.99966 0C8.99966 0 3.26172 0 1.79967 0C0.337614 0 -0.000304222 1.5 1.34969 3C2.69969 4.5 8.5003 9.5 8.99966 11C9.49902 12.5 8.99966 0 8.99966 0Z"
      fill="#f5f5f5"
    />
  </svg>
);

const REASON_OPTIONS = [
  {
    value: 'Language',
    label: 'Language',
  },
  {
    value: 'Offensive behaviour',
    label: 'Offensive behaviour',
  },
  {
    value: 'Spam',
    label: 'Spam',
  },
  {
    value: 'Fraud',
    label: 'Fraud',
  },
];

export const Comment: FC<CommentProps> = ({
  isMyComment,
  accountId,
  message,
  isCouncilUser,
  isReported,
  reportsCount,
  createdAt,
  onReport,
  onDelete,
  commentId,
}) => {
  const rootRef = useRef<HTMLLIElement>(null);
  const { t } = useTranslation();
  const isMounted = useMountedState();
  const [showReportModal] = useModal(ConfirmCommentActionModal);
  const [imagesUrls, setImagesUrls] = useState<string[]>([]);

  const handleReportAction = useCallback(async () => {
    const res = await showReportModal({
      title: 'Report on',
      text: 'Choose the reason for deleting the message ( 4 votes minimum for deletion)',
      options: REASON_OPTIONS,
    });

    if (res[0]) {
      onReport({ commentId, reason: res[0] as string });
    }
  }, [commentId, onReport, showReportModal]);

  const handleRemoveAction = useCallback(async () => {
    const res = await showReportModal({
      title: 'Delete message',
      text: 'Are you sure you want to delete the message from the chat? It will be deleted permanently. Choose the reason for deleting the message',
      options: REASON_OPTIONS,
    });

    if (res[0]) {
      onDelete(commentId, res[0] as string);
    }
  }, [commentId, onDelete, showReportModal]);

  function renderReportedMessage() {
    if (!reportsCount) {
      return null;
    }

    const counter = `${reportsCount}/4`;

    return (
      <div className={styles.reportedInfo}>
        {isReported
          ? `${t('comments.youReportedThisMessage')} ${counter}`
          : `${t('comments.thisMessageWasReported')} ${counter}`}
      </div>
    );
  }

  function renderActionButton(
    icon: IconName,
    action: () => void,
    tooltip: string
  ) {
    return (
      <div className={styles.commentControlButtonWrapper}>
        <ActionButton
          className={styles.commentControlButton}
          tooltipPlacement="top"
          iconClassName={styles.commentControlIcon}
          iconName={icon}
          onClick={action}
          tooltip={tooltip}
        />
      </div>
    );
  }

  useEffect(() => {
    if (!rootRef.current) {
      return;
    }

    const links = rootRef.current?.querySelectorAll(`.${styles.link}`);

    getImagesFromLinks(links).then(images => {
      if (images.length && isMounted()) {
        setImagesUrls(images);
      }
    });
  }, [isMounted]);

  return (
    <motion.li
      key={commentId}
      layout
      initial={{ opacity: 0 }}
      animate={{ opacity: 1 }}
      transition={{ duration: 0.3 }}
      className={cn(styles.root, {
        [styles.myComment]: isMyComment,
      })}
      ref={rootRef}
    >
      {isMyComment ? TICK_RIGHT : TICK_LEFT}
      {!isMyComment && <div className={styles.commentAuthor}>{accountId}</div>}
      <Linkify
        componentDecorator={decoratedHref => {
          return (
            <a
              key={decoratedHref}
              className={styles.link}
              href={decoratedHref}
              target="_blank"
              rel="noreferrer"
            >
              {imagesUrls.includes(decoratedHref) ? (
                // eslint-disable-next-line @next/next/no-img-element
                <img
                  src={decoratedHref}
                  alt={decoratedHref}
                  className={styles.inlineImage}
                />
              ) : (
                decoratedHref
              )}
            </a>
          );
        }}
      >
        <p>{message}</p>
      </Linkify>
      <div className={styles.time}>{formatISODate(createdAt, 'hh:mm a')}</div>
      <div
        className={cn(styles.commentControls, {
          [styles.reported]: isReported,
        })}
      >
        {isCouncilUser || isMyComment
          ? renderActionButton('buttonDelete', handleRemoveAction, 'Remove')
          : renderActionButton('commentBlock', handleReportAction, 'Report')}
        {renderReportedMessage()}
      </div>
    </motion.li>
  );
};
