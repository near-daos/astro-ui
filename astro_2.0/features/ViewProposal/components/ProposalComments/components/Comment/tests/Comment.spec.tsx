import { render } from 'jest/testUtils';

import { Comment } from 'astro_2.0/features/ViewProposal/components/ProposalComments/components/Comment';

jest.mock('next-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string): string => str,
    };
  },
}));

jest.mock('date-fns', () => {
  return {
    parseISO: () => 0,
    format: () => 'format mock',
  };
});

const onDelete = jest.fn();
const onReport = jest.fn();

describe('Comment', () => {
  it('Should render Comment component', () => {
    const { container } = render(
      <Comment
        accountId="james.testnet"
        commentId={1}
        isMyComment
        createdAt="2022-01-10T18:39:04.672Z"
        isCouncilUser={false}
        isReported={false}
        onDelete={onDelete}
        onReport={onReport}
        reportsCount={1}
        message="My name is Bond. James bond."
      />
    );

    expect(container).toMatchSnapshot();
  });
});
