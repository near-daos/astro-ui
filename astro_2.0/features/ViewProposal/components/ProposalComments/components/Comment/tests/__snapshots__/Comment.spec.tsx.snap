// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Should render Comment component 1`] = `
<div>
  <li
    class="root myComment"
    style="opacity: 0;"
  >
    <svg
      class="tickMark tickRight"
      fill="none"
      height="12"
      viewBox="0 0 10 12"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1.00034 0C1.00034 0 6.73828 0 8.20033 0C9.66239 0 10.0003 1.5 8.65031 3C7.30031 4.5 1.4997 9.5 1.00034 11C0.500977 12.5 1.00034 0 1.00034 0Z"
        fill="#F0F0F0"
      />
    </svg>
    <p>
      My name is Bond. James bond.
    </p>
    <div
      class="time"
    >
      format mock
    </div>
    <div
      class="commentControls"
    >
      <div
        class="commentControlButtonWrapper"
      >
        <div
          class="commentControlButton"
        >
          <button
            class="iconButton large btn"
            type="button"
          >
            <svg
              class="commentControlIcon"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="reportedInfo"
      >
        comments.thisMessageWasReported 1/4
      </div>
    </div>
  </li>
</div>
`;
